### ISSUE TITLE
Set schedule repeat bad request

### ISSUE THREAD

**Issue by markblokpoel** [lendy007/homeassistant-skodaconnect#120]
**Describe the bug**
When calling the `set_departure_schedule` with recurring option it fails with a Bad request error. I've tried to narrow down the cause, and I suspect that in `vehicle.py` line 628 should append weekdays (containing full strings) and not days (containing y/n), since I've seen the JSON returning full week days in an array and the data currently send is "['y', 'y', 'y']" when I set it to "yyynnnn".

```
data['timersSettings']['timers'][index]['recurringOn'].append(**week**days[num])
```

```
service: skodaconnect.set_departure_schedule
data:
  id: 1
  time: '06:30'
  enabled: true
  recurring: true
  days: 'yyynnnn'
  temp: 20
  climatisation: true
  device_id: 766f141621621dc3a48546ed99482058
```

**Latest working release**
1.0.67

**Debug logs**

```
2022-05-11 21:25:50 DEBUG (MainThread) [skodaconnect.connection] Request for "https://api.connect.skoda-auto.cz/api/v1/air-conditioning/TMBJB7NY6NF035846/timers" returned with status code [200]
2022-05-11 21:25:51 DEBUG (MainThread) [skodaconnect.connection] Request failed. Data: {'json': {'type': 'UpdateTimers', 'timersSettings': {'timers': [{'id': 1, 'enabled': True, 'type': 'RECURRING', 'time': '06:30', 'date': '2022-05-12', 'recurringOn': ['y', 'y', 'y']}, {'id': 2, 'enabled': False, 'type': 'ONE_OFF', 'time': '22:00', 'date': '2022-05-13'}]}}}, 
```

**Installation:**

 - Home Assistant variant: Core
 - Python version: 3.9.9
 - Python library versions: skodaconnect 1.1.19

**Additional context**
Using to connect with an Enyaq. Setting single departure schedule works. Let me know if further information / logs are required and if/how I can help test any possible fixes. Thanks a bunch for the awesome libraries!

**Comment by Farfar** [comment]
> **Describe the bug** When calling the `set_departure_schedule` with recurring option it fails with a Bad request error. I've tried to narrow down the cause, and I suspect that in `vehicle.py` line 628 should append weekdays (containing full strings) and not days (containing y/n), since I've seen the JSON returning full week days in an array and the data currently send is "['y', 'y', 'y']" when I set it to "yyynnnn".
> 
> ```
> data['timersSettings']['timers'][index]['recurringOn'].append(**week**days[num])
> ```
> 
> ```
> service: skodaconnect.set_departure_schedule
> data:
>   id: 1
>   time: '06:30'
>   enabled: true
>   recurring: true
>   days: 'yyynnnn'
>   temp: 20
>   climatisation: true
>   device_id: 766f141621621dc3a48546ed99482058
> ```
> 
> **Latest working release** 1.0.67
> 
> **Debug logs**
> 
> ```
> 2022-05-11 21:25:50 DEBUG (MainThread) [skodaconnect.connection] Request for "https://api.connect.skoda-auto.cz/api/v1/air-conditioning/TMBJB7NY6NF035846/timers" returned with status code [200]
> 2022-05-11 21:25:51 DEBUG (MainThread) [skodaconnect.connection] Request failed. Data: {'json': {'type': 'UpdateTimers', 'timersSettings': {'timers': [{'id': 1, 'enabled': True, 'type': 'RECURRING', 'time': '06:30', 'date': '2022-05-12', 'recurringOn': ['y', 'y', 'y']}, {'id': 2, 'enabled': False, 'type': 'ONE_OFF', 'time': '22:00', 'date': '2022-05-13'}]}}}, 
> ```
> 
> **Installation:**
> 
> * Home Assistant variant: Core
> * Python version: 3.9.9
> * Python library versions: skodaconnect 1.1.19
> 
> **Additional context** Using to connect with an Enyaq. Setting single departure schedule works. Let me know if further information / logs are required and if/how I can help test any possible fixes. Thanks a bunch for the awesome libraries!

Thanks, it seems you have found the culprit. I will take a look and fix for upcoming release. Might be a while though since I have limited time right now.
If you know how to fix this in code you can supply a pull request and it will save me some time :)

**Comment by markblokpoel** [comment]
Cool, happy I found a problem _and_ a solution, haha. I understand that your busy, I can create a PR with the changed code, but I'm not sure if I can run any tests before I push the changes.

**Comment by markblokpoel** [comment]
I changed the code and made a pull request: https://github.com/lendy007/skodaconnect/pull/55#issue-1234350305
I haven't been able to test it, since I don't know how to incorporate the adapted python code in my running home assistant instance.

**Comment by Farfar** [comment]
I found the culprit, I think. I will update when a new version with fix is posted.

**Comment by markblokpoel** [comment]
Amazing! Thanks so much! Let me know if I can help out with testing.

**Comment by Farfar** [comment]
> Amazing! Thanks so much! Let me know if I can help out with testing.

Please try out the latest beta, 1.0.70, and see if that solves the issue.

**Comment by Farfar** [comment]
@markblokpoel is this fixed with latest release?

**Comment by markblokpoel** [comment]
Tried it just now, but the entire integration isn't working. All services yield the following errors:

```
Logger: skodaconnect.connection
Source: custom_components/skodaconnect/__init__.py:357
Integration: Skoda Connect ([documentation](https://github.com/lendy007/homeassistant-skodaconnect/blob/main/README.md), [issues](https://github.com/lendy007/homeassistant-skodaconnect/issues))
First occurred: 11:30:11 (1 occurrences)
Last logged: 11:30:11

Failure during get request status: [VIN]
```

```
Logger: skodaconnect.vehicle
Source: custom_components/skodaconnect/__init__.py:357
Integration: Skoda Connect ([documentation](https://github.com/lendy007/homeassistant-skodaconnect/blob/main/README.md), [issues](https://github.com/lendy007/homeassistant-skodaconnect/issues))
First occurred: 11:30:11 (1 occurrences)
Last logged: 11:30:11
Exception encountered while waiting for request status: Failure during get request status: [VIN]
```

Any idea what causes this error?

**Comment by Farfar** [comment]
> Tried it just now, but the entire integration isn't working. All services yield the following errors:
> 
> ```
> Logger: skodaconnect.connection
> Source: custom_components/skodaconnect/__init__.py:357
> Integration: Skoda Connect ([documentation](https://github.com/lendy007/homeassistant-skodaconnect/blob/main/README.md), [issues](https://github.com/lendy007/homeassistant-skodaconnect/issues))
> First occurred: 11:30:11 (1 occurrences)
> Last logged: 11:30:11
> 
> Failure during get request status: [VIN]
> ```
> 
> ```
> Logger: skodaconnect.vehicle
> Source: custom_components/skodaconnect/__init__.py:357
> Integration: Skoda Connect ([documentation](https://github.com/lendy007/homeassistant-skodaconnect/blob/main/README.md), [issues](https://github.com/lendy007/homeassistant-skodaconnect/issues))
> First occurred: 11:30:11 (1 occurrences)
> Last logged: 11:30:11
> Exception encountered while waiting for request status: Failure during get request status: [VIN]
> ```
> 
> Any idea what causes this error?

Verify that the underlying library is updated. It should print the version during startup and it should be 1.1.25

**Comment by markblokpoel** [comment]
Yes, the correct version is loaded. I also get this specific message for the scheduler:

```
Logger: custom_components.skodaconnect
Source: custom_components/skodaconnect/__init__.py:344
Integration: Skoda Connect ([documentation](https://github.com/lendy007/homeassistant-skodaconnect/blob/main/README.md), [issues](https://github.com/lendy007/homeassistant-skodaconnect/issues))
First occurred: 11:34:28 (1 occurrences)
Last logged: 11:34:28
Failed to execute service call 'set_schedule' with data '<ServiceCall skodaconnect.set_departure_schedule (c:XXX): enabled=False, recurring=True, device_id=XXX, id=1, time=06:30, days=ynnnnnn, temp=20.0, climatisation=True>'
```

Strangely though the service calls are executed and processed correctly. It took a while for the changes to show up in the app (and later in the integration in HA). Do these errors imply anything significant? I could try to remove and re-add the integration, but I'd rather avoid that in case I get different entity names..

**Comment by Farfar** [comment]
> Yes, the correct version is loaded. I also get this specific message for the scheduler:
> 
> ```
> Logger: custom_components.skodaconnect
> Source: custom_components/skodaconnect/__init__.py:344
> Integration: Skoda Connect ([documentation](https://github.com/lendy007/homeassistant-skodaconnect/blob/main/README.md), [issues](https://github.com/lendy007/homeassistant-skodaconnect/issues))
> First occurred: 11:34:28 (1 occurrences)
> Last logged: 11:34:28
> Failed to execute service call 'set_schedule' with data '<ServiceCall skodaconnect.set_departure_schedule (c:XXX): enabled=False, recurring=True, device_id=XXX, id=1, time=06:30, days=ynnnnnn, temp=20.0, climatisation=True>'
> ```
> 
> Strangely though the service calls are executed and processed correctly. It took a while for the changes to show up in the app (and later in the integration in HA). Do these errors imply anything significant? I could try to remove and re-add the integration, but I'd rather avoid that in case I get different entity names..

The set schedule is a bit buggy. And the bugs are in API/cars so out of my control.
This however looks like something else and I'll have a look if I can reproduce or if it's a temporary issue.

**Comment by markblokpoel** [comment]
Thanks! I also noticed that the time set in the Skoda service is off by two hours (into the future). It also looks like the temperature is not correctly set. Anything I can do to help with debugging let me know!

**Comment by Farfar** [comment]
I believe that this might be fixed in 1.1.x release. I found some minor issues that might have caused issues like this.

**Comment by markblokpoel** [comment]
I thanks for coming back to this. I just checked with 1.1.4 and the 1 hour offset is still there.
